src/main/java/com/airbnb/kafka/NewStatsdMetricsReporter.java

```diff
@@ -0,0 +1,184 @@
+
+import java.util.EnumSet;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import com.timgroup.statsd.NonBlockingStatsDClient;
+import com.timgroup.statsd.StatsDClient;
+import com.timgroup.statsd.StatsDClientException;
+import com.yammer.metrics.Metrics;
+import com.yammer.metrics.core.Gauge;
+import com.yammer.metrics.core.MetricName;
+import com.yammer.metrics.core.MetricsRegistry;
+import com.yammer.metrics.reporting.AbstractPollingReporter;
+import org.apache.kafka.common.metrics.KafkaMetric;
+import org.apache.kafka.common.metrics.MetricsReporter;
+import org.slf4j.LoggerFactory;
+
+public class NewStatsdMetricsReporter implements MetricsReporter {
+  private static final org.slf4j.Logger log = LoggerFactory.getLogger(StatsDReporter.class);
+
+  static final String STATSD_REPORTER_ENABLED = "external.kafka.statsd.reporter.enabled";
+  static final String STATSD_HOST = "external.kafka.statsd.host";
+  static final String STATSD_PORT = "external.kafka.statsd.port";
+  static final String STATSD_METRICS_PREFIX = "external.kafka.statsd.metrics.prefix";
+  static final String POLLING_INTERVAL_SECS = "kafka.metrics.polling.interval.secs";
+  static final String STATSD_DIMENSION_ENABLED = "external.kafka.statsd.dimension.enabled";
+
+  private boolean enabled;
+  private final AtomicBoolean running = new AtomicBoolean(false);
+  private String host;
+  private int port;
+  private String prefix;
+  private long pollingPeriodInSeconds;
+  private EnumSet<Dimension> metricDimensions;
+  private StatsDClient statsd;
+  private MetricsRegistry registry;
+  private Map<String, KafkaMetric> kafkaMetrics;
+
+  private AbstractPollingReporter underlying = null;
+
+  public boolean isRunning() {
+    return running.get();
+  }
+
+  @Override
+  public void init(List<KafkaMetric> metrics) {
+    kafkaMetrics = new HashMap<String, KafkaMetric>();
+
+    if (enabled) {
+      startReporter(10);
+    } else {
+      log.warn("Reporter is disabled");
+    }
+  }
+
+  @Override
+  public void metricChange(final KafkaMetric metric) {
+    // Add new metrics in registry.
+    org.apache.kafka.common.MetricName metricName = metric.metricName();
+    String name = "kafka." + metricName.group() + "." + metricName.name();
+
+    StringBuilder strBuilder = new StringBuilder();
+
+    for (String key : metric.metricName().tags().keySet()) {
+      strBuilder.append(key).append(":").append(metric.metricName().tags().get(key)).append(",");
+    }
+
+    if (strBuilder.length() > 0) {
+      strBuilder.deleteCharAt(strBuilder.length() - 1);
+    }
+
+    if (!kafkaMetrics.containsKey(name)) {
+      Gauge<Double> gauge = new Gauge<Double>() {
+        @Override
+        public Double value() {
+          return metric.value();
+        }
+      };
+
+      registry.newGauge(NewStatsdMetricsReporter.class, name, strBuilder.toString() + "\n", gauge);
```

Comment:
* why a line ending in the name?
    * why not building a simpler and more explicit MetricName instance? `new MetricName(...)`